### PR TITLE
Add missing relative for getFileContents #5206

### PR DESCRIPTION
**Pull request by snoyberg**
Please include the following checklist in your PR:

* [X] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [X] Any changes that could be relevant to users have been recorded in the changelog.
* [X] The documentation has been updated, if necessary.
* [X] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

I confirmed with the repro in #5206

### CODE REVIEW

File: Cabal/Distribution/PackageDescription/Check.hs [hunk 1]
```diff
@@ -1830,7 +1830,7 @@ checkPackageFiles pkg root = checkPackageContent checkFilesIO pkg
       doesFileExist        = System.doesFileExist                  . relative,
       doesDirectoryExist   = System.doesDirectoryExist             . relative,
       getDirectoryContents = System.Directory.getDirectoryContents . relative,
-      getFileContents      = BS.readFile
+      getFileContents      = BS.readFile . relative
```
Comment (hvr) [inline Cabal/Distribution/PackageDescription/Check.hs]: obviously: indentation

**Comment by 23Skidoo** [comment]
Right, would be nice to have a test case.

**Comment by snoyberg** [comment]
I've updated with the formatting change requested. Unfortunately, I don't know enough about the test suite to know the right way to write a test case for this. I could do it by refactoring the set of functions and exposing `checkFilesIO`, but that would be an API change that I'm not sure would be accepted.

**Comment by hvr** [comment]
Why can't the repro case from #5206 turned into a regression test?

**Comment by snoyberg** [comment]
That may be the case. The missing bit is knowing how to add the appropriate package files into the repo. I see a bunch of parser tests, but I don't see any complete packages in the repo that could be hijacked for this. I don't mind adding such a minimal package, I'm just not sure where to put it.

**Comment by 23Skidoo** [comment]
Merged, thanks! A regression test would still be nice to have, but can be added later.

**Comment by hvr** [comment]
@23Skidoo note this was confusingly filed against 2.2 rather than `master`, so may need forward-porting

**Comment by 23Skidoo** [comment]
Yep, I cherry-picked it to `master`.

**Pull request merged**